// @nekazari/ui-kit/dist/index.js
// src/components/Button.tsx
import clsx from "clsx";
import { jsx } from "react/jsx-runtime";
var base = "inline-flex items-center justify-center rounded-md px-3 py-2 text-sm font-medium transition-colors focus:outline-none focus:ring-2 focus:ring-offset-2";
var variants = {
  primary: "bg-green-600 text-white hover:bg-green-700 focus:ring-green-500",
  secondary: "bg-gray-100 text-gray-900 hover:bg-gray-200 focus:ring-gray-400",
  ghost: "bg-transparent text-gray-700 hover:bg-gray-100 focus:ring-gray-300"
};
var Button = ({ variant = "primary", className, children, ...props }) => {
  return /* @__PURE__ */ jsx(
    "button",
    {
      className: clsx(base, variants[variant], className),
      ...props,
      children
    }
  );
};

// src/components/Card.tsx
import clsx2 from "clsx";
import { jsx as jsx2 } from "react/jsx-runtime";
var paddingMap = {
  none: "",
  sm: "p-3",
  md: "p-4",
  lg: "p-6"
};
var Card = ({ className, padding = "md", children, ...props }) => {
  return /* @__PURE__ */ jsx2(
    "div",
    {
      className: clsx2(
        "rounded-lg border border-gray-200 bg-white shadow-sm",
        paddingMap[padding],
        className
      ),
      ...props,
      children
    }
  );
};
export {
  Button,
  Card
};
