// @nekazari/sdk/dist/index.cjs
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var __publicField = (obj, key, value) => __defNormalProp(obj, typeof key !== "symbol" ? key + "" : key, value);

// src/index.ts
var index_exports = {};
__export(index_exports, {
  NKZClient: () => NKZClient,
  NekazariClient: () => NKZClient,
  NekazariI18nProvider: () => NekazariI18nProvider,
  Trans: () => import_react_i18next3.Trans,
  Translation: () => import_react_i18next3.Translation,
  changeLanguage: () => changeLanguage,
  getCurrentLanguage: () => getCurrentLanguage,
  getSupportedLanguages: () => getSupportedLanguages,
  i18n: () => import_i18next.default,
  initI18n: () => initI18n,
  useAuth: () => useAuth,
  useTranslation: () => import_react_i18next3.useTranslation,
  useViewer: () => useViewer,
  useViewerOptional: () => useViewerOptional
});
module.exports = __toCommonJS(index_exports);

// src/auth/useAuth.ts
var import_react = require("react");
function useAuth(session) {
  let hostAuthContext = null;
  if (typeof window !== "undefined") {
    try {
      hostAuthContext = window.__nekazariAuthContext;
      if (hostAuthContext) {
        console.log("[SDK useAuth] \u2705 Host auth context available");
      } else {
        console.warn("[SDK useAuth] \u26A0\uFE0F Host auth context not available in window.__nekazariAuthContext");
      }
    } catch (error) {
      console.warn("[SDK useAuth] Could not access host auth context:", error);
    }
  }
  const resolved = hostAuthContext ? {
    token: hostAuthContext.getToken?.() || hostAuthContext.token || void 0,
    tenantId: hostAuthContext.tenantId || hostAuthContext.getTenantId?.() || void 0,
    roles: hostAuthContext.user?.roles || hostAuthContext.roles || [],
    email: hostAuthContext.user?.email,
    username: hostAuthContext.user?.username
  } : session ?? {
    token: void 0,
    tenantId: void 0,
    roles: [],
    email: void 0,
    username: void 0
  };
  return (0, import_react.useMemo)(() => ({
    getToken: () => resolved.token,
    getTenantId: () => resolved.tenantId,
    hasRole: (role) => Array.isArray(resolved.roles) && resolved.roles.includes(role),
    hasAnyRole: (roles) => Array.isArray(resolved.roles) && roles.some((r) => resolved.roles.includes(r))
  }), [resolved.token, resolved.tenantId, resolved.roles]);
}

// src/i18n/provider.tsx
var import_react2 = require("react");
var import_react_i18next2 = require("react-i18next");

// src/i18n/config.ts
var import_i18next = __toESM(require("i18next"), 1);
var import_react_i18next = require("react-i18next");
var import_i18next_http_backend = __toESM(require("i18next-http-backend"), 1);
var import_i18next_browser_languagedetector = __toESM(require("i18next-browser-languagedetector"), 1);
var DEFAULT_CONFIG = {
  defaultLanguage: "es",
  fallbackLanguage: "es",
  supportedLanguages: ["es", "en", "ca", "eu", "fr", "pt"],
  loadPath: "/locales/{{lng}}/{{ns}}.json",
  namespaces: ["common"],
  debug: false
};
async function initI18n(config = {}) {
  const finalConfig = { ...DEFAULT_CONFIG, ...config };
  const storedLang = localStorage.getItem("language");
  const browserLang = navigator.language.split("-")[0];
  const detectedLang = storedLang || (finalConfig.supportedLanguages.includes(browserLang) ? browserLang : finalConfig.defaultLanguage);
  await import_i18next.default.use(import_i18next_http_backend.default).use(import_i18next_browser_languagedetector.default).use(import_react_i18next.initReactI18next).init({
    lng: detectedLang,
    fallbackLng: finalConfig.fallbackLanguage,
    supportedLngs: finalConfig.supportedLanguages,
    ns: finalConfig.namespaces,
    defaultNS: "common",
    // Backend configuration for loading translations
    backend: {
      loadPath: finalConfig.loadPath,
      // Allow cross-origin requests if needed
      crossDomain: false
    },
    // Language detection
    detection: {
      order: ["localStorage", "navigator"],
      caches: ["localStorage"],
      lookupLocalStorage: "language"
    },
    // React i18next options
    react: {
      useSuspense: true
    },
    // Interpolation
    interpolation: {
      escapeValue: false
      // React already escapes
    },
    // Debug mode
    debug: finalConfig.debug,
    // Performance optimizations
    load: "languageOnly",
    // Load 'es' instead of 'es-ES'
    cleanCode: true
    // Clean language codes
  });
  if (!storedLang) {
    localStorage.setItem("language", detectedLang);
  }
}
async function changeLanguage(language) {
  await import_i18next.default.changeLanguage(language);
  localStorage.setItem("language", language);
}
function getCurrentLanguage() {
  return import_i18next.default.language.split("-")[0] || "es";
}
function getSupportedLanguages() {
  return {
    es: "Espa\xF1ol",
    en: "English",
    ca: "Catal\xE0",
    eu: "Euskera",
    fr: "Fran\xE7ais",
    pt: "Portugu\xEAs"
  };
}

// src/i18n/provider.tsx
var import_react_i18next3 = require("react-i18next");
var import_jsx_runtime = require("react/jsx-runtime");
var NekazariI18nProvider = ({
  children,
  config,
  onLanguageChange
}) => {
  const [isInitialized, setIsInitialized] = (0, import_react2.useState)(false);
  const [initError, setInitError] = (0, import_react2.useState)(null);
  (0, import_react2.useEffect)(() => {
    let isMounted = true;
    const initialize = async () => {
      try {
        await initI18n(config);
        if (isMounted) {
          setIsInitialized(true);
        }
      } catch (error) {
        console.error("[NekazariI18n] Failed to initialize i18n:", error);
        if (isMounted) {
          setInitError(error instanceof Error ? error : new Error("Unknown error"));
          setIsInitialized(true);
        }
      }
    };
    initialize();
    const handleLanguageChanged = (lng) => {
      const lang = lng.split("-")[0];
      onLanguageChange?.(lang);
    };
    import_i18next.default.on("languageChanged", handleLanguageChanged);
    return () => {
      isMounted = false;
      import_i18next.default.off("languageChanged", handleLanguageChanged);
    };
  }, [config, onLanguageChange]);
  if (!isInitialized) {
    return /* @__PURE__ */ (0, import_jsx_runtime.jsx)("div", { className: "min-h-screen flex items-center justify-center bg-gray-50", children: /* @__PURE__ */ (0, import_jsx_runtime.jsxs)("div", { className: "text-center", children: [
      /* @__PURE__ */ (0, import_jsx_runtime.jsx)("div", { className: "animate-spin rounded-full h-12 w-12 border-b-2 border-green-600 mx-auto mb-4" }),
      /* @__PURE__ */ (0, import_jsx_runtime.jsx)("p", { className: "text-gray-600", children: "Loading translations..." })
    ] }) });
  }
  if (initError) {
    console.warn("[NekazariI18n] Continuing with i18n despite initialization error");
  }
  return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_react_i18next2.I18nextProvider, { i18n: import_i18next.default, children });
};

// src/viewer/useViewer.ts
function useViewer() {
  if (typeof window === "undefined") {
    throw new Error("useViewer can only be used in a browser environment");
  }
  const React2 = window.React;
  if (!React2 || !React2.useContext) {
    throw new Error("React is not available. Make sure the host application has exposed React globally.");
  }
  const ViewerContext = window.__nekazariViewerContextInstance;
  if (!ViewerContext) {
    throw new Error(
      "ViewerContext is not available. Make sure you are using this hook within the Nekazari platform and that the ViewerProvider is mounted in the host application."
    );
  }
  const context = React2.useContext(ViewerContext);
  if (context === void 0) {
    throw new Error("useViewer must be used within a ViewerProvider");
  }
  return context;
}
function useViewerOptional() {
  if (typeof window === "undefined") {
    return null;
  }
  const React2 = window.React;
  if (!React2 || !React2.useContext) {
    return null;
  }
  const ViewerContext = window.__nekazariViewerContextInstance;
  if (!ViewerContext) {
    return null;
  }
  const context = React2.useContext(ViewerContext);
  return context ?? null;
}

// src/api/client.ts
var NKZClient = class {
  constructor(options) {
    __publicField(this, "baseUrl");
    __publicField(this, "getToken");
    __publicField(this, "getTenantId");
    __publicField(this, "defaultHeaders");
    this.baseUrl = options.baseUrl.replace(/\/$/, "");
    this.getToken = options.getToken;
    this.getTenantId = options.getTenantId;
    this.defaultHeaders = options.defaultHeaders ?? {};
  }
  async request(path, init = {}) {
    const token = this.getToken?.();
    const tenant = this.getTenantId?.();
    const headers = {
      "Content-Type": "application/json",
      ...this.defaultHeaders,
      ...init.headers
    };
    if (token && !headers.Authorization) {
      headers.Authorization = `Bearer ${token}`;
    }
    if (tenant && !headers["X-Tenant-ID"]) {
      headers["X-Tenant-ID"] = tenant;
    }
    const url = `${this.baseUrl}${path.startsWith("/") ? "" : "/"}${path}`;
    const response = await fetch(url, {
      ...init,
      headers
    });
    if (!response.ok) {
      const text = await response.text().catch(() => "");
      throw new Error(`HTTP ${response.status} ${response.statusText}: ${text}`);
    }
    const contentType = response.headers.get("content-type") || "";
    if (contentType.includes("application/json")) {
      return response.json();
    }
    return response.text();
  }
  get(path, init = {}) {
    return this.request(path, { ...init, method: "GET" });
  }
  post(path, body, init = {}) {
    return this.request(path, {
      ...init,
      method: "POST",
      body: body !== void 0 ? JSON.stringify(body) : init.body
    });
  }
  put(path, body, init = {}) {
    return this.request(path, {
      ...init,
      method: "PUT",
      body: body !== void 0 ? JSON.stringify(body) : init.body
    });
  }
  patch(path, body, init = {}) {
    return this.request(path, {
      ...init,
      method: "PATCH",
      body: body !== void 0 ? JSON.stringify(body) : init.body
    });
  }
  delete(path, init = {}) {
    return this.request(path, { ...init, method: "DELETE" });
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  NKZClient,
  NekazariClient,
  NekazariI18nProvider,
  Trans,
  Translation,
  changeLanguage,
  getCurrentLanguage,
  getSupportedLanguages,
  i18n,
  initI18n,
  useAuth,
  useTranslation,
  useViewer,
  useViewerOptional
});
